ambry-store/src/main/java/com.github.ambry.store/StoreFindToken.java

```diff
@@ -0,0 +1,238 @@
+/**
+ * Copyright 2016 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.github.ambry.store;
+
+import com.github.ambry.utils.Utils;
+import java.io.DataInputStream;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.util.UUID;
+
+
+/**
+ * The StoreFindToken is an implementation of FindToken.
+ * It is used to provide a token to the client to resume
+ * the find from where it was left previously. The StoreFindToken
+ * maintains a offset to track entries within the journal. If the
+ * offset gets outside the range of the journal, the storekey and
+ * indexstartoffset that refers to the segment of the index is used
+ * to perform the search. This is possible because the journal is
+ * always equal or larger than the writable segment.
+ */
+public class StoreFindToken implements FindToken {
+  private static final byte[] ZERO_LENGTH_ARRAY = new byte[0];
+
+  /**
+   * The type of the store token.
+   */
+  public enum Type {
+    Uninitialized, JournalBased, IndexBased
+  }
+
+  private Type type;
+  private Offset offset;
+  private StoreKey storeKey;
+  private UUID sessionId;
+  private long bytesRead;
+
+  private static final short VERSION = 1;
+  private static final int VERSION_SIZE = 2;
+  private static final int TYPE_SIZE = 2;
+  private static final int SESSION_ID_SIZE = 4;
```

Comment:
this is actually `SESSION_ID_LENGTH_SIZE`, isn't it? And 2 bytes should suffice really (if you wanted to make that change, now would be a good time).